### ISSUE TITLE
BigBlueButton not working

### ISSUE THREAD

**Issue by aelalfey** [chamilo/chamilo-lms#3321]
I installed BBB on my server and everything seems to be working fine on the server Greenlight working with no issues but when i enter configuration settings inside Chamilo for the video conference plugin and try to run a session i get this page https://ibb.co/H7fC7D9

Also when refreshing the session link sessions appears at the Chamilo page but when i try to click end session i get the same blank 500 error mentioned above. https://ibb.co/H2SPNMW

Here is my configuration for Video plugin https://ibb.co/t8nCNX9

Note: i tried removing "/bigbluebutton/" from the server URL with no luck. same error.

Please advice.

Thank you in advance

**Comment by ywarnier** [comment]
Hi,
Please have a little patience for Chamilo 1.11.12 which comes with a large number of improvements and bugfixes on the BigBlueButton plugin.
Alternatively, you can download the current 1.11.x branch from here, unzip it, extract the plugin/bbb/ directory and replace yours with this new version (take a backup first as this is an untested procedure), then uninstall/reinstall the plugin. This might work if you're in a hurry, but it is not an official solution.

**Comment by aelalfey** [comment]
Hi ywarnier,

Thank you for your reply. 

I did give it a try BBB plugin "V 2.8.1" with my current Chamilo "V 1.11.10" and BBB server running "V 2.2.17" when i activated the plugin and entered the BBB configuration page to add BBB server information and hit the "save" button, the page reloaded with "changes saved" message but all the fields that i already entered in the configuration page returns empty as if i didn't fill it out in the first place. 

So i guess i have to wait till the official release of Chamilo 1.11.12.

Thank you once more.

**Comment by ywarnier** [comment]
@aelalfey could you share the start of the URL line of "bbb-conf --secret" (without the "Secret:" line, obviously) so as to understand how your BBB server is configured and what the difference is with ours?

Someone reported (can't find the issue number right now) an issue with a lack of protocol in the URL returned by the BBB API. If this is the case, then reformulating the URL on the BBB server side might be enough.

**Comment by ywarnier** [comment]
I was refering to #2706  note by by @caos30

**Comment by aelalfey** [comment]
@ywarnier Thank you once more for your reply. 

URL: https://vc-mac.org/bigbluebutton/

I tested what @caos30 mentioned but it didn't work too. 

Every-time i click on "Enter the videoconference" it opens a blank page

"This page isn’t working
domain.org is currently unable to handle this request.
HTTP ERROR 500"

**Comment by caos30** [comment]
@aelalfey what i did to find my problem was this, quite easy: using FTP or SSH modify this file of the chamilo plugin (on the chamilo server)
```
/plugin/bbb/lib/bbb_api.php
```
and insert at top these lines:
```
ini_set('display_errors',1);
ini_set('display_startup_errors',1);
error_reporting(E_ALL ^ E_DEPRECATED ^ E_USER_DEPRECATED );
```
Then, try to open again a conference room from Chamilo and you will see that the PHP errors who are causing the error 500 are displayed on the web browser!!! Please, paste here ;-)

**Comment by aelalfey** [comment]
@caos30  @ywarnier 

Thank you for your reply here is the error that i got:

`Warning: SimpleXMLElement::__construct(): Entity: line 6: parser error : Opening and ending tag mismatch: hr line 5 and body in /var/www/vhosts/mac-eg.org/learn/plugin/bbb/lib/bbb_api.php on line 82

Warning: SimpleXMLElement::__construct(): </body> in /var/www/vhosts/mac-eg.org/learn/plugin/bbb/lib/bbb_api.php on line 82

Warning: SimpleXMLElement::__construct(): ^ in /var/www/vhosts/mac-eg.org/learn/plugin/bbb/lib/bbb_api.php on line 82

Warning: SimpleXMLElement::__construct(): Entity: line 7: parser error : Opening and ending tag mismatch: body line 3 and html in /var/www/vhosts/mac-eg.org/learn/plugin/bbb/lib/bbb_api.php on line 82

Warning: SimpleXMLElement::__construct(): </html> in /var/www/vhosts/mac-eg.org/learn/plugin/bbb/lib/bbb_api.php on line 82

Warning: SimpleXMLElement::__construct(): ^ in /var/www/vhosts/mac-eg.org/learn/plugin/bbb/lib/bbb_api.php on line 82

Warning: SimpleXMLElement::__construct(): Entity: line 8: parser error : Premature end of data in tag html line 1 in /var/www/vhosts/mac-eg.org/learn/plugin/bbb/lib/bbb_api.php on line 82

Warning: SimpleXMLElement::__construct(): in /var/www/vhosts/mac-eg.org/learn/plugin/bbb/lib/bbb_api.php on line 82

Warning: SimpleXMLElement::__construct(): ^ in /var/www/vhosts/mac-eg.org/learn/plugin/bbb/lib/bbb_api.php on line 82

Fatal error: Uncaught Exception: String could not be parsed as XML in /var/www/vhosts/mac-eg.org/learn/plugin/bbb/lib/bbb_api.php:82 Stack trace: #0 /var/www/vhosts/mac-eg.org/learn/plugin/bbb/lib/bbb_api.php(82): SimpleXMLElement->__construct('<html>\r\n<head><...') #1 /var/www/vhosts/mac-eg.org/learn/plugin/bbb/lib/bbb_api.php(170): BigBlueButtonBN->_processXmlResponse('https://vc-mac....') #2 /var/www/vhosts/mac-eg.org/learn/plugin/bbb/lib/bbb.lib.php(389): BigBlueButtonBN->createMeetingWithXmlResponseArray(Array) #3 /var/www/vhosts/mac-eg.org/learn/plugin/bbb/start.php(80): bbb->createMeeting(Array) #4 {main} thrown in /var/www/vhosts/mac-eg.org/learn/plugin/bbb/lib/bbb_api.php on line 82`

**Comment by caos30** [comment]
Ok, now, in the same file, search this line of code (in my version is on line #70):
```
curl_setopt( $ch, CURLOPT_URL, $url );
```
and insert above this line this other line to see on browser which is the URL that the plugin is calling to:
```
echo "<p>url: $url</p>";die();
```
Please, paste the result.

**Comment by aelalfey** [comment]
@caos30 

url: vc-mac.org/bigbluebutton/api/getMeetingInfo?meetingID=15efbf373032ac8.14833639&password=url_1_1&checksum=password

**Comment by aelalfey** [comment]
@caos30 

When i opened the link above in browser i got this message 

"This XML file does not appear to have any style information associated with it. The document tree is shown below.
<response>
<returncode>FAILED</returncode>
<messageKey>notFound</messageKey>
<message>We could not find a meeting with that meeting ID</message>
</response>"

**Comment by caos30** [comment]
exactly... main friend: you're suffering the same thing i suffered yesterday. I think that you didn't do what i explained on my issue that @ywarnier has referred to you. This probably should fixed your problem. Let me explain: this bbb_api.php file is calling an url WITHOUT HTTPS:// on the start of the URL !!

Make this and tell me if your problem get fixed:
- remove the first lines you added to display errors on that file
- remove this last line with ```echo "...";die()```;
- instead of that line __echo__ put this other:
```
$url = str_replace('https://https://','https://','https://'.$url);
```
in other words, above the line:
```
curl_setopt( $ch, CURLOPT_URL, $url );
```
Then try to open again a conference room ;-)

**Comment by aelalfey** [comment]
@caos30 

Thanks a million. I did work. I was putting the line after "curl_setopt( $ch, CURLOPT_URL, $url );" that is why it didn't work the first time.

Thank you once more.

**Comment by caos30** [comment]
I'm glad to be useful (Bicentennial man, dixit).

**Comment by caos30** [comment]
My pologies @aelalfey , reviewing now my other post on the other issue where i sahred this solution i made a mistake with the number of line where to insert the patch!!! You're right: i should use an explanation like i used today here: specifying above which line to put it!

Thanks to you too.

**Comment by Nazka99** [comment]
thanks, @caos30 I had the same problem and thanks to your help I managed to solve it

**Comment by ywarnier** [comment]
I believe this patch about the url and protocol solves the issue too: https://github.com/chamilo/chamilo-lms/commit/1cf87448d11459cc682568724e2b1b7d8f8d2189
